common:
  speed:
    slowly: &SLOWLY
      speed: 0.2
      accel: 0.2
      timeout: 10
    normally: &NORMALLY
      speed: 0.4
      accel: 0.4
      timeout: 6.
    quickly: &QUICKLY
      speed: 1.
      accel: 1.
      timeout: 4.

  tolerance:
    normal_tolerance: &NORMAL_TOLERANCE
      tolerance_joints: [ 0.06, 0.006, 0.006, 0.006, 0.1, 0.1 ]
    bigger_tolerance: &BIGGER_TOLERANCE
      tolerance_joints: [ 0.01, 0.01, 0.01, 0.01, 0.2, 0.2 ]

  gripper:
    open_gripper: &OPEN_GRIPPER
      state: false
    close_gripper: &CLOSE_GRIPPER
      state: true

  arm:
    normal_home: &NORMAL_HOME
      joints: [ 0.008, 0.025, 0.024, -0.6264, 0., 0. ]
      <<: *NORMALLY
      <<: *NORMAL_TOLERANCE

  gimbal:
    sky_gimbal: &SKY_POS
      frame: gimbal_base
      position: [ -0.2, -0.0008, 1 ]
    back_gimbal: &BACK_POS
      frame: gimbal_base
      position: [ -2 ,-0.1, -15 ]
    big_stone_gimbal: &BIG_STONE_POS
      frame: gimbal_base
      position: [ 2, 0.01, 0.1 ]
    walking_pos: &WALKING_POS
      frame: gimbal_base
      position: [ -2 ,0.01, -15 ]

steps_list:
  ################### COMMON STEP####################
  DELETE_SCENE:
    - step: "delete_scene"
  ################### HOME ####################
  NORMAL_HOME0:
    - step: "approach"
      arm:
        <<: *NORMAL_HOME

  ################### GIMBAL ####################
  SKY_GIMBAL:
    - step: "open gripper"
      gimbal:
        <<: *SKY_POS
  WALK_GIMBAL:
    - step: "open gripper"
      gimbal:
        <<: *WALKING_POS

  BIG_STONE_GIMBAL:
    - step: "open gripper"
      gimbal:
        <<: *BIG_STONE_POS

  BACK_GIMBAL:
    - step: "gimbal"
      gimbal:
        <<: *BACK_POS

  ################### GRIPPER ####################
  OPEN_GRIPPER:
    - step: "open gripper"
      gripper:
        <<: *OPEN_GRIPPER

  CLOSE_GRIPPER:
    - step: "close gripper"
      gripper:
        <<: *CLOSE_GRIPPER
  ################### CHASSIS ####################
  CHASSIS_FORWARD:
    - step: "forward"
      chassis:
        frame: base_link
        position: [ 0.3, 0. ]
        yaw: 0.0
        chassis_tolerance_position_: 0.1
        chassis_tolerance_angular_: 0.01

  CHASSIS_BACKWARD:
    - step: "backward"
      chassis:
        frame: base_link
        position: [ -0.3, 0. ]
        yaw: 0.0
        chassis_tolerance_position_: 0.1
        chassis_tolerance_angular_: 0.01

  CHASSIS_LEFT:
    - step: "left"
      chassis:
        frame: base_link
        position: [ 0., 0.5 ]
        yaw: 0.0
        chassis_tolerance_position_: 0.1
        chassis_tolerance_angular_: 0.01

  CHASSIS_RIGHT:
    - step: "right"
      chassis:
        frame: base_link
        position: [ 0., -0.5 ]
        yaw: 0.0
        chassis_tolerance_position_: 0.1
        chassis_tolerance_angular_: 0.01

  CHASSIS_ROTATE:
    - step: "rotate"
      chassis:
        frame: base_link
        position: [ 0., 0. ]
        yaw: 3.14
        chassis_tolerance_position_: 0.1
        chassis_tolerance_angular_: 0.01
